### ISSUE TITLE
Weather Underground plugin broken due to API change (v2 no longer supported?)

### ISSUE THREAD

**Issue by peerk72** [domoticz/domoticz#3304]
Since last week (2019 June 6th) the WU sensors aren't update anymore in my domoticz.
I get the same error message as occured earlier this year.

Before June 6th it was still working as expected so i assume something changed recently.

When i look at the url that's used in the domoticz code (line 156 just before the error that is thrown) I see v2 of the api. In a lot of other places in the code it is allready changed to v3

sURL << "https://api.weather.com/v2/pws/observat ... stationId=" << szLoc << "&format=json&units=m&apiKey=" << m_APIKey;
bool bret;
std::string szURL=sURL.str();
bret=HTTPClient::GET(szURL,sResult);
if (!bret)
{
_log.Log(LOG_ERROR,"Wunderground: Error getting http data! (Check API key!)");
return;
}

I assume v2 isn't supported anymore (by wunderground) and we should use v3 also in this place.
I think we should use next url

sURL << "https://api.weather.com/v3/location/poi ... US&apiKey=" << m_APIKey <<"&pws=" << szLoc << "&format=json"

This url works in the browser (i see some json result, btw v2 url as used in the current code doesn't give any results at all)
I don't know how to change this myself. Is there somebody who could have a look at it?

Thanks in advance!!

**Comment by gizmocuz** [comment]
I assume your URL's above are corrupt as you probably did not used URL tags.
I tested the original code with a weather station here (IHILVERS97) and i got normal results
changing v2 to v3 did return an empty result

I noticed you also posted in another issue and on the forum

https://github.com/domoticz/domoticz/issues/3127

https://www.domoticz.com/forum/viewtopic.php?f=12&t=27246&start=20#p216719

Using the link from the forum did also not return any response

Did you enter a StationID or a latitude/longitude in the hardware setup ?

**Comment by peerk72** [comment]
i see now that the stationid (ITETERIN12) i used no longer is available 
(i changed to another stationid (ITETERIN13) and now it's working again)

that's probably why other requests (like the ones with v3 in the url) and with the same apikey worked 
and this probably didn't work

so i'll close this ticket,

PS is it better to use a stationid or to use latitude/longitude to avoiud these issues?

**Comment by gizmocuz** [comment]
Thanks for the feedback! If you want to have local weather I would use the Geo coordinates... Specially because a station can indeed change

**Comment by peerk72** [comment]
when i use geo-coordinates i again get the same error message
as long as i use a valid stationid it works 

could it be the longitude / latitude i enter isn't recognized as a coordinate and the domoticz code tries to evaluate it as a normal location/stationid and then the error is thrown.

**Comment by gizmocuz** [comment]
What did you enter@

**Comment by peerk72** [comment]
I tried it with next coordinates:
51.61,4.82

I also tried using place (Teteringen, Netherlands) this worked previously also but somehow works now either. 

When I look at the wunderground web page and search with these coordinates or place I get results but not what I would expect. They return weather information from Eindhoven and that's by far not the nearest weather station. I think I have to stick to using a station ID as location and switch to another station when that one fails in future.
